src/simulation/navigation/pinholeCamera/pinholeCamera.h

```diff
@@ -0,0 +1,100 @@
+/*
+ ISC License
+
+ Copyright (c) 2016, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+
+#ifndef PINHOLE_CAMERA_H
+#define PINHOLE_CAMERA_H
+
+#include <Eigen/Dense>
+#include <vector>
+#include <string>
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/msgPayloadDefC/EphemerisMsgPayload.h"
+#include "architecture/msgPayloadDefC/SCStatesMsgPayload.h"
+#include "architecture/msgPayloadDefC/AccessMsgPayload.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/utilities/astroConstants.h"
+#include "architecture/utilities/bskLogging.h"
+
+/*! @brief Pinhole camera class */
+class PinholeCamera:  public SysModel {
+public:
+    PinholeCamera();
+    ~PinholeCamera();
+    void UpdateState(uint64_t CurrentSimNanos);
+    void Reset(uint64_t CurrentSimNanos);
+    void readInputMessages();
+    void writeOutputMessages(uint64_t CurrentClock);
+    
+    void addLandmark(Eigen::Vector3d& pos, Eigen::Vector3d& normal);
+    void processBatch(Eigen::MatrixXd rBatch_BP_P, Eigen::MatrixXd mrpBatch_BP, Eigen::MatrixXd eBatch_SP_P);
+    
+private:
+    void processInputs();
+    void loopLandmarks();
+    bool checkLighting(Eigen::Vector3d nLmk);
+    bool checkFOV(Eigen::Vector2i pLmk, Eigen::Vector3d rLmk, Eigen::Vector3d nLmk);
+    Eigen::Vector2i computePixel(Eigen::Vector3d rLmk);
+
+public:
+    /* Pinhole camera properties */
+    double f; //!< [m] camera focal length
+    double FOVx; //!< [rad] horizontal field of view
+    double FOVy; //!< [rad] vertical field of view
+    double nxPixel; //!< [-] number of horizontal pixels
+    double nyPixel; //!< [-] number of vertical pixels
+    double wPixel; //!< [m] pixel width
+    Eigen::Matrix3d dcm_CB; //!< [-] dcm from body to camera
+    
+    /* Module outputs */
+    int nvisibleLmk; //!< [-] number of visible landmarks
+    Eigen::VectorXi isvisibleLmk; //!< [-] flag telling if a landmark is visible
+    Eigen::MatrixXi pixelLmk; //!< [-] pixels for landmarks
+        
+    /* Landmark distribution */
+    int n; //!< [-] number of landmarks
```

Comment:
shouldn't this be a private variable?  It is set in reset by your module, not the use.  Please check that only variables are public that should be set by the user or are need to be accessible outside of this class.

Reply:
Yes, it could be. I see no issues to have this one private. I may push this a little bit oftentimes as I like to access them to have the number at hand (e.g. field of view angles are computed internally but public)